pkg/toml/util.go

```diff
@@ -0,0 +1,48 @@
+package toml
+
+import (
+	"os"
+	"fmt"
+	"github.com/naoina/toml"
+	"github.com/SeerUK/tid/pkg/types"
+)
+
+// BoltDatabaseFilename is the name of the database file name on disk.
+const TomlConfigFilename = "config.toml"
+
+// Open opens the configuration file or it creates it if it doesn't exist already.
+func Open(tidDir string) (types.TomlConfig, error) {
+	config := types.NewTomlConfig()
+
+	// Make the `path` if it does not exist.
+	err := os.MkdirAll(tidDir, os.ModePerm)
+	if err != nil {
+		return config, err
+	}
+
+	var configFilePath = fmt.Sprintf("%s/%s", tidDir, TomlConfigFilename)
+
+	f, err := os.OpenFile(configFilePath, os.O_RDONLY|os.O_CREATE, 0644)
+
+	if err != nil {
+		panic(err)
```

Comment:
I notice this method can return an `error` type. Generally, methods which do this should not `panic()` but instead allow the error to be handled elsewhere by returning it. This may require some rethinking of the return type `types.TomlConfig` as well though